bin/desi_qso_mgii_afterburner

```diff
@@ -0,0 +1,250 @@
+        DESI_TARGET (str): name of DESI_TARGET for the wanted version of the target selection
+        clobber (bool): overwrite the fits file defined by filename ?
+    Returns:
+        None
+    """
+    # Ok we cannot use dataframe.to_records() car les coeffs/c_lines sont sauvegarder sous forme de list de type objet et ne peux pas etre convertit ..
+    data = np.zeros(dataframe.shape[0], dtype=[('TARGETID', 'i8'), ('RA','f8'), ('DEC','f8'), ('Z_RR','f8'), ('ZERR', 'f4'), ('IS_QSO_MGII', '?'),
+                                               (DESI_TARGET, 'i8'), ('SPECTYPE', 'U10'), #('COEFFS', ('f4', 10)),
+                                               ('DELTA_CHI2', 'f4'), ('A', 'f4'), ('SIGMA', 'f4'), ('B', 'f4'),
+                                               ('VAR_A', 'f4'), ('VAR_SIGMA', 'f4'), ('VAR_B', 'f4')])
+
+    data['TARGETID'] = dataframe['TARGETID']
+    data['RA'] = dataframe['RA']
+    data['DEC'] = dataframe['DEC']
+    data['Z_RR'] = dataframe['Z_RR']
+    data['ZERR'] = dataframe['ZERR']
+    data['IS_QSO_MGII'] = dataframe['IS_QSO_MGII']
+    data[DESI_TARGET] = dataframe[DESI_TARGET]
+    data['SPECTYPE'] = dataframe['SPECTYPE']
+    #data['COEFFS'] = np.array([np.array(dataframe['COEFFS'][i]) for i in range(dataframe.shape[0])])
+
+    data['DELTA_CHI2'] = dataframe['DELTA_CHI2']
+    data['A'] = dataframe['A']
+    data['SIGMA'] = dataframe['SIGMA']
+    data['B'] = dataframe['B']
+    data['VAR_A'] = dataframe['VAR_A']
+    data['VAR_SIGMA'] = dataframe['VAR_SIGMA']
+    data['VAR_B'] = dataframe['VAR_B']
+
+    fits = fitsio.FITS(filename, 'rw', clobber=clobber)
+    fits.write(data, extname='MGII')
+    if clobber:
+        log.info(f'OVERWRITE the file : {filename}')
+    else:
+        log.info(f'EXPAND the file : {filename}')
+    fits.close()
+
+    return
+
+
+if __name__ == "__main__":
+
+    args = collect_argparser()
+
+    # Param for the MgII fitter see desispec/py/desispec/mgii_afterburner.py for additional informations
+    param_mgii_fitter = {'lambda_width':args.lambda_width, 'max_sigma':args.max_sigma, 'min_sigma':args.min_sigma,
+                         'min_deltachi2':args.min_deltachi2, 'min_signifiance_A':args.min_signifiance_A, 'min_A':args.min_A}
+
+
+    if os.path.isfile(args.coadd) & os.path.isfile(args.zbest):
+        # open best fit file generated by redrock
+        zbest_file = fitsio.FITS(args.zbest)
+        zbest = zbest_file[1]
+        fibermap = zbest_file[2]
+        index_sorted = np.argsort(fibermap['TARGETID'][:500]) # in zbest targetid are sorted, it is not the case in the fibermap
+
+        if args.target_version == 'SV1':
+            DESI_TARGET = 'SV1_DESI_TARGET'
+        elif args.target_version == 'SV3':
+            DESI_TARGET = 'SV3_DESI_TARGET'
+        elif args.target_version == 'MAIN':
+            DESI_TARGET = 'DESI_TARGET'
+        else:
+            log.error("**** CHOOSE CORRECT TARGET_VERSION FLAG (SV1 / SV3 / MAIN) ****")
+            sys.exit()
```

Comment:
To simplify incorporating this into the pipeline, let's use `desitarget.targets.main_cmx_or_sv(fibermap)` to auto-derive which flavor this input is and which column name to use (with the added benefit that also supports SV2 and will support any other flavors of targeting that may appear in the future).

Reply:
ok